invoker/core/src/main/java/com/google/cloud/functions/invoker/runner/Invoker.java

```diff
@@ -225,8 +227,52 @@ ClassLoader getFunctionClassLoader() {
     return functionClassLoader;
   }
 
+  /**
+   * This will start the server and wait (join) for function calls.
+   * To start the server inside a unit or integration test, use {@link #startTestServer()} instead.
+   *
+   * @see #stopServer()
+   * @throws Exception
+   */
   public void startServer() throws Exception {
-    Server server = new Server(port);
+    startServer(true);
```

Comment:
Could you put something like this first?

```java
    if (server != null) {
      throw new IllegalStateException("Server already started");
    }
```

Reply:
Would it be better to do it inside the `startServer(boolean)` method instead of here ?

Reply:
Yes, you are right of course.